packages/twenty-server/src/engine/core-modules/ask-ai/ask-ai.service.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable } from '@nestjs/common';
+
+import { ChatOpenAI } from '@langchain/openai';
```

Comment:
Do you use https://platform.openai.com/docs/guides/text-generation/json-mode and https://platform.openai.com/docs/guides/function-calling? You probably should?

Reply:
I understand why we would eventually not want to use function calling eventually to avoid coupling too much with Open AI? But JSON mode can only be positive?

Reply:
I'll check this out if non-SQL repsonses from the LLM keep being an issue after iterating the prompt!